controllers/storageoscluster/node-manager_operand.go

```diff
@@ -95,12 +96,19 @@ func (c *NodeManagerOperand) ReadyCheck(ctx context.Context, obj client.Object)
 	nodeManagerDep := &appsv1.Deployment{}
 	key := client.ObjectKey{Name: nmDeploymentName, Namespace: obj.GetNamespace()}
 	if err := c.client.Get(ctx, key, nodeManagerDep); err != nil {
-		log.Info("node-manager not ready")
+		log.Info("node-manager fetch error")
 		return false, err
 	}
 
-	if nodeManagerDep.Status.AvailableReplicas > 0 {
-		log.V(4).Info("Found available replicas more than 0", "availableReplicas", nodeManagerDep.Status.AvailableReplicas)
+	// Get node replicas to compare.
+	replicas, err := c.getNodeReplicas(context.Background(), obj, log)
```

Comment:
Minor detail, "getNodeManagerReplicas" would make it clear what we are working with, there's the "node" image as well so it can be confusing IMO

Reply:
Nope, this is node replicas not node manager replicas.